### ISSUE TITLE
No/Wrong build instructions for Linux (probably other platforms as well)

### ISSUE THREAD

**Issue by x4121** [TASVideos/desmume#60]
It's not clear how to build this project from just looking at the repo (base README has no real value, couldn't find any README that has build instructions).

Link to http://wiki.desmume.org/index.php?title=Installing_DeSmuME_from_source in 'desmume/README' is outdated.

'Latest Stable Release' from http://desmume.org/download/ cannot be built from source with gcc.

Could you please provide build instructions?

**Comment by intact** [comment]
Instal dependencies and use:

```shell
cd desmume/src/frontend/posix/
./autogen.sh
./configure
make
```

**Comment by zeromus** [comment]
Those steps were for the latest code on github.

"cannot be built from source with gcc" -- sounds unlikely to me, as there are many distros with that package.

**Comment by x4121** [comment]
> sounds unlikely to me, as there are many distros with that package.

Well.. most distros also maintain own patches/fixes in downstream projects, so it's not THAT unlikely.
At least I'm getting errors about shift expressions on negative numbers and non-'constant int' values beeing assigned as enum values.
```
MMU_timing.h:158:28: error: left operand of shift expression ‘(-1 << 11)’ is negative [-fpermissive]
  enum { TAGMASK = (u32)(~0 << TAGSHIFT) };
...
MMU_timing.h:159:7: error: enumerator value for ‘BLOCKMASK’ is not an integer constant
  enum { BLOCKMASK = ((u32)~0 >> (32 - TAGSHIFT)) & (u32)(~0 << BLOCKSIZESHIFT) };
```

> Can you please provide actual information about what's out of date about the build instructions?

- It's still refering to the svn repo
- It instructs to run `./autogen.sh` and `./configure` in the project root

**Comment by zeromus** [comment]
wiki fixed.
compile error well-discussed on google. i'm sure it's been fixed in git since 0.9.11

**Comment by x4121** [comment]
> i'm sure it's been fixed in git since 0.9.11

It is, building from master works fine. However I guess a lot of people will try to build from release versions first, since they tend to be more stable/reliable than master branches.
